src/lib.rs

```diff
@@ -83,11 +83,10 @@ where
 impl Context {
     /* Constructors */
     /// Get a context structure from a built artifact.
-    fn from_artifact(artifact: &Artifact) -> Result<Self, failure::Error> {
+    fn from_artifact(metadata: Metadata, artifact: &Artifact) -> Result<Self, failure::Error> {
         // Get target from artifact. Ideally, the artifact should really contain
         // the target triple. Sadly, it doesn't. So as an approximation, we
         // extract it from the filename path.
```

Comment:
The place of this comment doesn't seem appropriate after removing the code after it. We should probably move or rework it.